### PR TITLE
refactor(ava/data): migrate dataset and improve dataset

### PR DESCRIPTION
**Pull request by pddpd**
- rename dataAnalysis to analysis
- migrate and improve dw/dataset files
- add tests for dataset

### CODE REVIEW

File: packages/ava/src/data/utils/isType/isType.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+export function isString(value: unknown): value is string {
+  return typeof value === 'string';
+}
+
+export function isNumber(value: unknown): value is number {
+  if (typeof value === 'number') return true;
+  return false;
+}
+
+export function isNumberString(value: unknown): value is string {
```
Comment (Esorakouki) [inline packages/ava/src/data/utils/isType/isType.ts:16]: test case: '.1', '00000', '00000.1'

File: packages/ava/src/data/utils/isType/isType.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  }
+  return false;
+}
+
+export function isInteger(value: unknown): value is number {
+  if (typeof value === 'number') return Number.isInteger(value);
+  return false;
+}
+
+export function isIntegerString(value: unknown): value is string {
```
Comment (Esorakouki) [inline packages/ava/src/data/utils/isType/isType.ts:46]: test case: '000001'

File: packages/ava/src/data/utils/isType/isType.ts (hunk 1 above)
Comment (pddpd) [reply packages/ava/src/data/utils/isType/isType.ts:16]: have added, tests passed

File: packages/ava/src/data/utils/isType/isType.ts (hunk 2 above)
Comment (pddpd) [reply packages/ava/src/data/utils/isType/isType.ts:46]: have added, tests passed

File: packages/ava/src/data/utils/isType/isType.ts (hunk 1 above)
Comment (pddpd) [reply packages/ava/src/data/utils/isType/isType.ts:16]: https://github.com/antvis/AVA/pull/492/files#diff-3f4aedd26848ef6985fdb24a469349ccdf6e313c7afd75b86552232240e6d087R33-R57

**Review by Esorakouki — APPROVED** [review]
LG

**Pull request merged**